### ISSUE TITLE
Segfault in extractError function

### ISSUE THREAD

**Issue by denisenkom** [gstreamer-java/gst1-java-core#280]
Hi all,

I encountered SIGSEGV error in Gst.extractError function, here is full error report file:
[hs_err_pid32632.log](https://github.com/gstreamer-java/gst1-java-core/files/13801575/hs_err_pid32632.log)

When a JNA function is invoked with a `Structure.ByReference` parameter, JNA automatically reads the contents of this structure upon completion of the native function call. This step is crucial for propagating any modifications made by the native function back to a corresponding Java class that mirrors the native structure. The implementation responsible for this process can be found [here](https://github.com/java-native-access/jna/blob/master/src/com/sun/jna/Function.java#L383).

The seg fault happens when `extractError` function calls `g_error_free` function which releases memory used by `GErrorStruct` structure.
Since `g_error_free` function releases memory used by the passed structure and structure is passed by reference, JNA attempts to read from a memory that has been released. This leads to "use-after-free" bug.

I made a fix to prevent JNA from reading structure content after call to `g_error_free` function: https://github.com/gstreamer-java/gst1-java-core/pull/279

**Comment by denisenkom** [comment]
Versions:
```
Java: Corretto-16.0.2.7.1
OS: Darwin 22.6.0
JNA: 5.14.0
gst1-java-core: 1.4.0
```

**Comment by neilcsmith-net** [comment]
Thanks!  I'll take a proper look in the new year.  I'm not sure whether to fix it in the way you have here, although it's probably safer to do so.  It might indicate issues in a few other places.  I need to check if anywhere else is trying to use `volatile` to indicate not auto-syncing the field, whereas it only switches off auto-writing.

The other option is probably to use `setAutoSynch(false)` in the constructors of this struct.  The code path via `extractError` also needs checking for tests, because I think it's missing.

It's interesting that this issue has not been reported before, which might be due to memory allocator differences, or just that it's rare(?)

**Comment by denisenkom** [comment]
I think an ideal solution would be to have an annotation on that parameter in that function to indicate that this is an input only parameter but JNA does not have such annotation afaik.

Yeah it is interesting, I am able to reproduce this issue with 100% reproducibility rate. And able to confidently confirm that the fix in my pull request works.

**Comment by denisenkom** [comment]
A friendly reminder. Does this PR require changes?

**Comment by neilcsmith-net** [comment]
I think it's good, although a test that fail before application and passes after would make it perfect!

However, at the moment we need to make a decision on whether we're going to do any further releases of this library.  I will merge if and when we decide to do so.

Thanks!

**Comment by denisenkom** [comment]
On Mac I am able to reproduce a test crash before fix and a test pass after fix with the following test:
```java
    @Test
    public void testParseLaunchThrowingError() {
        ArrayList<GError> errors = new ArrayList<GError>();
        assertThrows(GstException.class, () -> {
            Gst.parseLaunch("fakesrcc ! fakesinkk", errors);
        });
    }
```
On Linux I wasn't able to reproduce crash with this test with debug heap enabled so far.

While trying to reproduce I found similar memory issues. I get a crash when running tests on Linux with debug heap enabled (with or without fix applied). I was able to isolate crashing tests to EventTest and QueryTest.
This command causes test to crash:
```shell
LD_PRELOAD=/lib/x86_64-linux-gnu/libc_malloc_debug.so.0 MALLOC_CHECK_=3 mvn -Dtest='EventTest,QueryTest' test
```
While this command passes:
```shell
LD_PRELOAD=/lib/x86_64-linux-gnu/libc_malloc_debug.so.0 MALLOC_CHECK_=3 mvn -Dtest='!EventTest,!QueryTest' test
```

Wanted to share those intermediate results. I will spend a bit more time trying to reproduce the original issue on Linux.

**Comment by OliverTED** [comment]
This still seems to be an issue in the official builds. If I understand correctly, this bug raises a segfault (and crashes the JVM) if you start gstreamer with a malformed pipeline description. A [fix](https://github.com/gstreamer-java/gst1-java-core/commit/b61e65139015ff1340b4fac0d290e424d276218c) was proposed by @denisenkom in his repository. 

**Please merge this fix.**

**Comment by neilcsmith-net** [comment]
@OliverTED yes, because there has not been a release in 4.5 years.  It's not as simple as just merging this either.  There's a bunch of other work to do to get this project back into a mergeable and releaseable state.  If I get some time, I'll try and get an update release done in a couple of months - other priorities for now.
